index.js

```diff
@@ -21,7 +21,7 @@ const customConfig = {};
 
 const getConfig = function getConfig() {
   const args = [].slice.call(arguments);
-  return Object.assign.apply(null, [{}, defaults, customConfig].concat(args));
+  return _.merge.apply(_, [{}, defaults, customConfig].concat(args));
```

Comment:
You are probably thinking of `clone` which is shallow, and has a corresponding `cloneDeep`.